src/Evolution/Systems/Cce/Equations.hpp

```diff
@@ -0,0 +1,1153 @@
+ * which appear as arguments for spin-weighted derivatives, which are primarily
+ * radial (y) derivatives
+ * - `cached_sw_derivatives` : The set of spin-weighted derivatives used in CCE
+ *   computations
+ * - `secondary_cce_values` : The set of values which are useful during CCE
+ *   computations and are derived from Bondi quantities or boundary inputs, but
+ *   are not updated in between hypersurface steps
+ * - `temporary_cache` : the set of temporary spin-weighted scalars used to
+ *   cache reused intermediate values during individual integrand evaluations.
+ * - `bondi_integrands` : needs only contain `IntegrandTag`, and functions as
+ *   the output of the computation.
+ *
+ * For each individual computation, the list of tags which it accesses from the
+ * input variables can be quite lengthy, so there is the helper type alias
+ * `tags_needed_for_integrand_computation`, which is a tmpl::list of the tags
+ * needed for the computation. The type alias takes two template argument, the
+ * first is the `IntegrandTag` computation which it describes, and the second is
+ * a `TagsCategory` type which describes which of the input variables arguments
+ * it is relevant for. The association between inputs and the `TagsCategory` is:
+ * - `bondi_and_sw_cache` : `BondiAndSWCache`
+ * - `cached_sw_derivatives` : `Derivatives`
+ * - `secondary_cce_values` : `Secondary`
+ * - `temporary_cache` : `Temporary`
+ *
+ * Additional mathematical details for each of the computations can be found in
+ * the template specializations of this struct.
+ */
+template <typename IntegrandTag>
+struct ComputeBondiIntegrand;
+
+/*!
+ * \brief Computes the integrand (right-hand side) of the equation which
+ * determines the radial (y) dependence of the Bondi quantity \f$\beta\f$.
+ *
+ * \details The quantity \f$\beta\f$ is defined via the Bondi form of the
+ * metric:
+ * \f[
+ * ds^2 = - \left(e^{2 \beta} (1 + r W) - r^2 h_{AB} U^A U^B\right) du^2 - 2
+ * e^{2 \beta} du dr - 2 r^2 h_{AB} U^B du dx^A + r^2 h_{A B} dx^A dx^B. \f]
+ * Additional quantities \f$J\f$ and \f$K\f$ are defined using a spherical
+ * angular dyad \f$q^A\f$:
+ * \f[ J \equiv h_{A B} q^A q^B, K \equiv h_{A B} q^A
+ * \bar{q}^B.\f]
+ * See \cite Bishop1997ik \cite Handmer2014qha for full details.
+ *
+ * In the compactified coordinate \f$ y \equiv 1 - 2 R/ r\f$, where \f$R\f$
+ * is determined by the worldtube initial data, the equation which
+ * determines \f$\beta\f$ on a surface of constant \f$u\f$ given \f$J\f$ on
+ * the same surface is, \f[\partial_y (\beta) = \frac{1}{8} (-1 + y)
+ * \left(\partial_y (J) \partial_y
+ * (\bar{J}) -  \frac{(\partial_y (J \bar{J}))^2}{4 K^2}\right). \f]
+ */
+template <>
+struct ComputeBondiIntegrand<Tags::IntegrandFor<Tags::Beta>> {
+  template <typename BondiAndSWCacheTagList, typename DerivativeTagList,
+            typename IntegrandTagList, typename SecondaryTagList,
+            typename TemporaryTagList>
+  void evaluate(
```

Comment:
As discussed at the call yesterday, a comment of ours for some reason didn't make it into the review. We suggested to make this conform to the existing `db::mutate_apply` mechanism, see also #1413. Let's talk in a hangout if anything is unclear.